pkg/cmd/server/api/types.go

```diff
@@ -117,6 +117,11 @@ type MasterConfig struct {
 	// PauseControllers instructs the master to not automatically start controllers, but instead
 	// to wait until a notification to the server is received before launching them.
 	PauseControllers bool
+	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
+	// a lease before controllers start and renewing it within a number of seconds defined by this value.
+	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
+	// omitted) and controller election can be disabled with -1.
```

Comment:
For people who want to manually control controller launching it's a good
option.  Leasing has a cost to etcd (because of continual renewals).  We
should consider removing it in 3.1.

On Fri, Aug 14, 2015 at 4:02 PM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/api/types.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37113818>:
>
> > @@ -117,6 +117,11 @@ type MasterConfig struct {
> >  	// PauseControllers instructs the master to not automatically start controllers, but instead
> >  	// to wait until a notification to the server is received before launching them.
> >  	PauseControllers bool
> > +	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
> > +	// a lease before controllers start and renewing it within a number of seconds defined by this value.
> > +	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
> > +	// omitted) and controller election can be disabled with -1.
>
> Why do we want to allow pausing in addition to leasing? I can image all
> sorts of contrived scenarios, but are any likely? Seems like a
> "try-to-lease-now" trigger would be more useful for the fast failover cases.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37113818>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift